src/merkle/smt/mod.rs

```diff
@@ -743,19 +779,58 @@ impl<const DEPTH: u8, K: Serializable, V: Serializable> Serializable for Mutatio
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
         target.write(self.old_root);
         target.write(self.new_root);
-        self.node_mutations.write_into(target);
-        self.new_pairs.write_into(target);
+
+        let removals: Vec<_> = self
+            .node_mutations
+            .iter()
+            .filter(|(_, value)| matches!(value, NodeMutation::Removal))
+            .map(|(key, _)| key)
+            .collect();
+        let additions: Vec<_> = self
+            .node_mutations
+            .iter()
+            .filter_map(|(key, value)| match value {
+                NodeMutation::Addition(node) => Some((key, node)),
+                _ => None,
+            })
+            .collect();
+
+        target.write_u16(
+            removals.len().try_into().expect("Number of items to remove must fit in u16"),
+        );
+        target.write_many(removals);
+
+        target
+            .write_u16(additions.len().try_into().expect("Number of items to add must fit in u16"));
+        target.write_many(additions);
+
+        target.write_u16(self.new_pairs.len() as u16);
+        target.write_many(&self.new_pairs);
```

Comment:
Good catch! I expected that this serialization is needed only for miden-node, where the number of mutations is currently limited to `MAX_ACCOUNTS_PER_BLOCK`, which is 65536 now. But it was my mistake, because even this value needs 17 bits, so I need to update the code, thank you!

Reply:
Regarding using `usize` here, I avoid of using data fields which size may vary in different systems. For example, this can be serialized on system with 32-bit usize, but deserialized on system with 64-bit usize and data will be corrupted. I think, `u32` should be enough here for all possible cases.

Reply:
`usize` is actually serialized using platform-independent variable-length encoding. So, it should be fine here.

Reply:
Ah, I didn't know that winter-utils implement it platform-independently, thank you! I will update the implementation then.